CRM/Admin/Form/Extensions.php

```diff
@@ -199,16 +199,18 @@ public function postProcess() {
     }
 
     if ($this->_action & CRM_Core_Action::UPDATE) {
-      $result = civicrm_api('Extension', 'download', [
-        'version' => 3,
-        'key' => $this->_key,
+      $downloads = CRM_Extension_System::singleton()->getBrowser()->findDownloads([$this->_key]);
+
+      $qd = new CRM_Extension_QueueDownloader();
+      // FIXME: It would be nice to accept a list of exts and/or use transitive dependencies (Manager::findInstallRequirements()).
+      $runner = new CRM_Queue_Runner([
+        'title' => $qd->getTitle(),
+        'queue' => $qd->fillQueue($qd->createQueue(), $downloads),
```

Comment:
This calls `fillQueue()` with some default arguments, `$upgradeDb: true, $cleanup: true`. It's debatable which policy is preferable.

* IMHO, for the most part, it's kind of silly to force a user to click-around to do those things manually -- so I think this should be an ordinary use-case.
* OTOH, I can totally respect folks who want to have more fine-grained control.
* I wouldn't mind some kind of policy mechanism, but not sure what. A global setting? (`ext_auto_upgrade_db`?) Or perhaps a [new checkbox in the UI](https://github.com/user-attachments/assets/93f174d3-dd94-4362-9dc9-ac35aafdcb42)? And what's the priority - is that a blocker for the PR, or is it a future improvement?